### PR TITLE
[stable/mysql] add option to run sql script post container start

### PR DESCRIPTION
**Pull request by dkuerner**
- add ConfigMap for sql initialization files
- mount initialization files to /docker-entrypoint-initdb.d if present

**What this PR does / why we need it**:
Enables to mount sql files which will be executed after the container started.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
@viglesiasce

### CODE REVIEW

**Comment by timstoop** [comment]
It looks like these will be executed on every start or does the container entrypoint make sure it doesn't? If it does execute on every start, you might want to explain clearly in the docs that the code should be idempotent, ie. not break things when it's already in place.

**Comment by dkuerner** [comment]
@timstoop Good point. Just double-checked the behaviour. The entrypoint.sh shell script does make sure that files (*.sh, *.sql, *.sql.gz) mounted into /docker-entrypoint-initdb.d/ are run exactly once when the container is created initially and not on container restarts.

File: stable/mysql/README.md [hunk 1]
```diff
@@ -115,3 +116,18 @@ configurationFiles:
   mysql_custom.cnf: |-
     [mysqld]
 ```
+
+## MySQL initialization files
+
+The [MySQL](https://hub.docker.com/_/mysql/) image accepts *.sh, *.sql and *.sql.gz files at the path `/docker-entrypoint-initdb.d`. 
+These files are being run exactly once for container initialization and ignored on follwing container restarts.
```
Comment (timstoop) [inline stable/mysql/README.md]: *following

File: stable/mysql/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: mysql
-version: 0.3.7
+version: 0.3.8
```
Comment (timstoop) [inline stable/mysql/Chart.yaml]: You cna bump the minor of the version, as this adds new functionality.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: stable/mysql/README.md (hunk 1 above)
Comment (dkuerner) [reply stable/mysql/README.md]: done

File: stable/mysql/Chart.yaml (hunk 2 above)
Comment (dkuerner) [reply stable/mysql/Chart.yaml]: done

**Comment by timstoop** [comment]
/assign @viglesiasce 
Due to being the registered maintainer of this Chart.

File: stable/mysql/templates/configmap.yaml [hunk 3]
```diff
@@ -2,10 +2,22 @@
...
 metadata:
-  name: {{ template "mysql.fullname" . }}
+  name: {{ template "mysql.fullname" . }}-configuration
 data:
 {{- range $key, $val := .Values.configurationFiles }}
   {{ $key }}: |-
 {{ $val | indent 4}}
 {{- end }}
+{{- end -}}
+---
```
Comment (unguiculus) [inline stable/mysql/templates/configmap.yaml]: Please create a separate file for this.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/mysql/templates/configmap.yaml (hunk 3 above)
Comment (dkuerner) [reply stable/mysql/templates/configmap.yaml]: Hi @unguiculus,
thanks for your input.

- Done splitting configmaps into seperate files.
- Synced my fork
- rebased branch onto master

Puzzled however, why pull-charts-e2e is failing.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**